executor/new_builder.go

```diff
@@ -140,6 +142,18 @@ func (b *executorBuilder) buildNewTableScan(v *plan.NewTableScan) Executor {
 			Columns:     v.Columns,
 			ranges:      v.Ranges,
 		}
+		var remained []expression.Expression
+		e.where, remained = b.newConditionExprToPBExpr(client, v.FilterConditions, v.Table)
```

Comment:
If you build range from selection, then get filter conditions from parent selection.